chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/SplitActionPermissions.java

```diff
@@ -1,21 +1,77 @@
 package gov.healthit.chpl.permissions.domains.product;
 
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import gov.healthit.chpl.dao.CertifiedProductDAO;
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.dto.CertifiedProductDetailsDTO;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.dto.ProductDTO;
+import gov.healthit.chpl.entity.developer.DeveloperStatusType;
+import gov.healthit.chpl.exception.EntityRetrievalException;
 import gov.healthit.chpl.permissions.domains.ActionPermissions;
 
 @Component("productSplitActionPermissions")
 public class SplitActionPermissions extends ActionPermissions {
 
+    @Autowired
+    private DeveloperDAO developerDao;
+
+    @Autowired
+    private CertifiedProductDAO certifiedProductDao;
+
     @Override
     public boolean hasAccess() {
-        return getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()
-                || getResourcePermissions().isUserRoleAcbAdmin();
+        return false;
     }
 
     @Override
     public boolean hasAccess(Object obj) {
-        return false;
+        if (!(obj instanceof ProductDTO)) {
+            return false;
+        } else if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
+            return true;
+        } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
+            try {
+                ProductDTO dto = (ProductDTO) obj;
+                if (isDeveloperActive(dto.getDeveloperId())) {
+                    return doesCurrentUserHaveAccessToAllOfDevelopersListings(dto.getDeveloperId());
+                } else {
+                    return false;
+                }
+            } catch (Exception e) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    private boolean doesCurrentUserHaveAccessToAllOfDevelopersListings(Long developerId) {
+        List<CertifiedProductDetailsDTO> cpDtos = certifiedProductDao.findByDeveloperId(developerId);
+        for (CertifiedProductDetailsDTO cpDto : cpDtos) {
+            if (!isAcbValidForCurrentUser(cpDto.getCertificationBodyId())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isDeveloperActive(Long developerId) {
```

Comment:
I see code checking whether a developer is active in at least 3 places.. can you combine to a common helper-type class or superclass?